src/integrationtest/java/com.microsoft.aad.msal4j/AuthorizationCodeIT.java

```diff
@@ -155,6 +164,30 @@ private void assertAcquireTokenAAD(User user){
         Assert.assertEquals(user.getUpn(), result.account().username());
     }
 
+    private void assertAcquireTokenAADWithClaims(User user){
+
+        PublicClientApplication pca;
+        try {
+            pca = PublicClientApplication.builder(
+                    user.getAppId()).
+                    authority(cfg.organizationsAuthority()).
+                    build();
+        } catch(MalformedURLException ex){
+            throw new RuntimeException(ex.getMessage());
+        }
+
+        String authCode = acquireAuthorizationCodeAutomatedWithClaims(user, pca);
+        IAuthenticationResult result = acquireTokenAuthorizationCodeFlow(
+                pca,
+                authCode,
+                Collections.singleton(cfg.graphDefaultScope()));
+
+        Assert.assertNotNull(result);
+        Assert.assertNotNull(result.accessToken());
+        Assert.assertNotNull(result.idToken());
+        Assert.assertEquals(user.getUpn(), result.account().username());
+    }
```

Comment:
I think you should be able to assert that the id token that gets returned has the claims that you requested.